lib/blog_helpers.rb

```diff
@@ -51,25 +55,29 @@ def related_articles(article, limit = 3)
 
   private
 
-  def sepcontent
+  def content
     data_directory = if @app.data[:sepcontent].nil? || ENV["DATA_DIR"].eql?("sample")
       :sample
     else
       :sepcontent
     end
-    @sepcontent ||= @app.data[data_directory]
+    @content ||= @app.data[data_directory]
   end
 
   def article_entries
-    if sepcontent && sepcontent.articles
-      structurize(sepcontent.articles)
+    @article_entries ||= if content && content.articles
+      structurize(content.articles)
         .sort_by(&:date)
         .reverse
     end
   end
 
+  def page_entries
+    @page_entries ||= content.pages.map { |page| OpenStruct.new(page[1]) } if content && content.pages
+  end
+
   def category_entries
-    structurize(sepcontent.categories) if sepcontent && sepcontent.categories
+    @category_entries ||= structurize(content.categories) if content && content.categories
```

Comment:
@skatkov that's right, sometimes when you stare at your own code long enough you elaborate a special type of blindness when readability doesn't matter any more :D Thanks for the advice though!